internal/command/meta_providers.go

```diff
@@ -376,22 +376,33 @@ func providerFactory(meta *providercache.CachedProvider) providers.Factory {
 			return nil, err
 		}
 
-		// store the client so that the plugin can kill the child process
 		protoVer := client.NegotiatedVersion()
-		switch protoVer {
-		case 5:
-			p := raw.(*tfplugin.GRPCProvider)
-			p.PluginClient = client
-			p.Addr = meta.Provider
-			return p, nil
-		case 6:
-			p := raw.(*tfplugin6.GRPCProvider)
-			p.PluginClient = client
-			p.Addr = meta.Provider
-			return p, nil
-		default:
-			panic("unsupported protocol version")
+		p, err := initializeProviderInstance(raw, protoVer, client, meta.Provider)
+		if err != nil {
+			panic(err)
```

Comment:
Could we use a sentinel error here? I'd prefer not to blanket panic on an error received from a function,

That is, introduce an errUnsuportedProtocolVersion global, and then we can return it from the function, and check for it here (and panic only for it).

I know right now it's the only error case, but someone could presumably add a non-panic error condition to the function and we could miss it here.

Reply:
Done in f37c60439e37c2a56a4547e460acbde35b0aee0b